pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/PicklistQuestionDao.java

```diff
@@ -105,15 +108,69 @@ default void insertGroup(long questionId, PicklistGroupDef group, int order, lon
      * @return the option ids
      */
     default List<Long> insertOptions(long questionId, List<PicklistOptionDef> options, long revisionId) {
-        List<Long> optionIds = new ArrayList<>();
+        boolean nonTextTemplateFound = options.stream()
+                .anyMatch(o -> o.getTooltipTemplate() != null && o.getTooltipTemplate().getTemplateType() != TemplateType.TEXT);
+
+        if (nonTextTemplateFound) {
+            new DaoException("Only TEXT template type is supported for tooltips");
+        }
+
+        // Let's try to get all the template for all the options in a single batch
+        List<Template> templateList = Stream.of(
+                options.stream().map(o -> o.getOptionLabelTemplate()),
+                options.stream().map(o -> o.getTooltipTemplate()),
+                options.stream().map(o -> o.getDetailLabelTemplate()),
+                options.stream().map(o -> o.getNestedOptionsLabelTemplate())
+        ).flatMap(i -> i).collect(toList());
+
```

Comment:
Line up all the templates for all the options into a single list in preparation for the their batch insertion.